AmplifyPlugins/DataStore/AWSDataStoreCategoryPlugin/Sync/SubscriptionSync/ReconcileAndLocalSave/ReconcileAndLocalSaveOperation.swift

```diff
@@ -214,8 +214,17 @@ class ReconcileAndLocalSaveOperation: AsynchronousOperation {
 
     private func saveDeleteMutation(storageAdapter: StorageEngineAdapter, remoteModel: RemoteModel) {
         log.verbose(#function)
-        guard let modelType = ModelRegistry.modelType(from: remoteModel.model.modelName) else {
-            let error = DataStoreError.invalidModelName(remoteModel.model.modelName)
+        let modelInstance = remoteModel.model.instance
+        let modelName: String
+        if let jsonModel = modelInstance as? JSONValueHolder,
+           let modelNameFromJson = jsonModel.jsonValue(for: "__typename") as? String {
```

Comment:
Do we have the `__typename` field modeled as a protocol or typename somewhere? It feels a little weird to suddenly be diving into conflict resolution payload field names here

Reply:
Removed this line as per our discussion and uses ModelSchema directly. I will see if I can do the same in another pr for the other usage of `__typename`.